### ISSUE TITLE
False positive for 'no-template-curly-in-string'

### ISSUE THREAD

**Issue by sziraqui** [eslint/eslint#14000]
**Tell us about your environment**

* **ESLint Version:** 7.17.0
* **Node Version:** 12.19.1
* **npm Version:** 6.14.9

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?** `@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```
{
    "env": {
        "node": true,
        "es6": true,
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 2020,
        "sourceType": "module"
    },
    "plugins": ["@typescript-eslint"]
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
const x = '`${ foo: 1, bar: 0}`';
```

```bash
npx -q eslint . --ext .ts --fix
```

**What did you expect to happen?**
No error

**What actually happened? Please include the actual, raw output from ESLint.**
```
error  Unexpected template string expression  no-template-curly-in-string
```

**Are you willing to submit a pull request to fix this bug?**
I will try to, maybe this weekend.

**Comment by anikethsaha** [comment]
Thanks for the report. 
I can reproduce this in [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdGVtcGxhdGUtY3VybHktaW4tc3RyaW5nOiBcImVycm9yXCIqL1xuXG5jb25zdCB4ID0gJ2AkeyBmb286IDEsIGJhcjogMH1gJzsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLXJlZGVjbGFyZSI6Mn0sImVudiI6e319fQ==)
Currently the rule searches for `${...}` inside a string literal so that's why the regex is returning `true` [here](https://github.com/eslint/eslint/blob/master/lib/rules/no-template-curly-in-string.js#L34)

I think this rule should ignore string literals having `${...}` between two back tick (`).  But this kind of cases are rare. 

IMO, disabling this line for using eslint's comment directives would be a better choice.

**Comment by mdjermanovic** [comment]
I think this is working as expected, this rule warns on what looks like `${...}` syntax.

If `${...}` is intentional in a string literal, you can use `eslint-disable` comments.

**Comment by nzakas** [comment]
Agree, this is working as intended. The point of the rule is to flag `${` in regular strings and it is doing so here.
